sklearn/linear_model/tests/test_sag.py

```diff
@@ -826,3 +826,24 @@ def test_multinomial_loss_ground_truth():
                         [-0.903942, +5.258745, -4.354803]])
     assert_almost_equal(loss_1, loss_gt)
     assert_array_almost_equal(grad_1, grad_gt)
+
+
+@pytest.mark.parametrize("solver", ["sag", "saga"])
+def test_sag_classifier_raises_error(solver):
+    # Following #13316, the error handling behavior changed in cython sag. This
+    # is simply a non-regression test to make sure numerical errors are
+    # properly raised.
+
+    # Train a classifier on a simple problem
+    rng = np.random.RandomState(42)
+    X, y = make_classification(random_state=rng)
+    clf = LogisticRegression(solver=solver, random_state=rng, warm_start=True)
+    clf.fit(X, y)
```

Comment:
Maybe record `clf.n_iter_` here an check below that `"at epoch #%d" % clf.n_iter_ + 1` appears in the error message.

Reply:
actually the reported n_iter in the error message raised by the cython code when warm start is enabled cannot be correct (as in consistent with `clf.n_iter_`) because of the way the Cython code is structured. I don't want to make mix concerns in this PR, let's keep simple.